packages/react-mongo/react-mongo.ts

```diff
@@ -24,14 +32,31 @@ const useSubscriptionClient = (
     // it stops the inner one.
     const computation = Tracker.nonreactive(() => (
       Tracker.autorun(() => {
-        setSubscription( Meteor.subscribe( name, ...args ) )
+        subscription.current = Meteor.subscribe( name, ...args )
+        // @ts-ignore this is just an internal thing
+        subscription.current.deps = { name, args }
       })
     ))
 
     return () => computation.stop()
   }, [name, ...args])
 
-  return subscription
+  return useCallback(
+    () => {
+      if (subscription.current) {
+        // @ts-ignore
+        const { deps } = subscription.current
+        if (deps.name === name && deps.args === args) {
```

Comment:
I think you don't need to manually track and compare deps here, you can just set subscription.current back to null in the effect cleanup phase ?

Reply:
Well, the thing is, this all happens _after_ the render. So it will already be too late by then.

Reply:
The cleanup phase runs after the render, but before the ready() callback you return gets executed I think ?
Then by the time it gets executed, subscription.current has gone back to null and you can just return false ?
(not 100% sure off the top of my head, but worth a try I think)

More importantly, this code still has the same drawback as before : by not running the reactive subscription.current.ready() in some cases, the function we build here won't be reactive later ?

I have something in mind, need to try it first :-)

Reply:
> The cleanup phase runs after the render, but before the ready() callback you return gets executed I think ?
Then by the time it gets executed, subscription.current has gone back to null and you can just return false ?

Nope. Tried 10 different ways, even with `useState`, but every time the first `ready()` call still references the old subscription.